analyzers/tests/SonarAnalyzer.UnitTest/Wrappers/TypeInfoExtensionsTest.cs

```diff
@@ -0,0 +1,169 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using StyleCop.Analyzers.Lightup;
+using NullabilityInfo = StyleCop.Analyzers.Lightup.NullabilityInfo;
+using NullableAnnotation = StyleCop.Analyzers.Lightup.NullableAnnotation;
+using NullableFlowState = StyleCop.Analyzers.Lightup.NullableFlowState;
+
+namespace SonarAnalyzer.UnitTest.Wrappers
+{
+    [TestClass]
+    public class TypeInfoExtensionsTest
+    {
+        [TestMethod]
+        public void NullabilityInfoFromShimEqualsOriginal()
+        {
+            var code = @"
+#nullable enable
+public class C
+{
+    public void M()
+    {
+        object? o = null;
+        o.ToString();
+    }
+}
+";
+            var (tree, semanticModel) = TestHelper.CompileCS(code);
+            var identifier = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Last(x => x.NameIs("o")); // o in o.ToString()
+            var typeInfo = semanticModel.GetTypeInfo(identifier);
+
+            var nullability = typeInfo.Nullability();
```

Comment:
To make it clear what is Roslyn and what is shim
```suggestion
            var nullabilityShim = typeInfo.Nullability();
```

Reply:
Same for all below